cpp/src/parquet/column_chunker.cc

```diff
@@ -0,0 +1,778 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "parquet/column_chunker.h"
+#include <cmath>
+#include <string>
+#include <vector>
+#include "arrow/array.h"
+#include "arrow/util/logging.h"
+#include "parquet/level_conversion.h"
+
+namespace parquet {
+namespace internal {
+
+constexpr uint64_t GEAR_HASH_TABLE[8][256] = {
```

Comment:
These constants are generated by a python script, copy-pasting is simple but I am open to suggestions.

Reply:
Sure, what would be the right place for this? Just in the same directory?

Reply:
Added the script.